alibi_detect/utils/random.py

```diff
@@ -0,0 +1,65 @@
+from contextlib import contextmanager
+import random
+import numpy as np
+import os
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow
+
+if has_tensorflow:
+    import tensorflow as tf
+if has_pytorch:
+    import torch
+
+
+def set_seed(seed):
+    """
+    Sets the Python, NumPy, TensorFlow and PyTorch random seeds (if installed).
+    """
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
```

Comment:
Not sure if relevant, but what was the deal with using new-style `numpy` random number generators?

Reply:
AFAIK, this shouldn't affect the numpy generators, in that they don't appear to be affected by the `np.random.seed()` at all. A very crude example:

```python
from alibi_detect.utils.random import set_seed, get_seed

rng = np.random.default_rng()
set_seed(0)
a = rng.random()
set_seed(0)
b = rng.random()
assert a == b  # False

rng = np.random.default_rng(0)
a = rng.random()
rng = np.random.default_rng(0)
b = rng.random()
assert a == b  # True
```

I've changed `utils.random` to private for now, since going forward we might want to more deeply consider how we can move from the legacy `numpy` RNG to generators.